source/Locales/ScottLocale/ScottLocalesOfScottDomains.lagda

```diff
@@ -0,0 +1,816 @@
+
+\end{code}
+
+The module:
+
+\begin{code}
+
+open import Locales.ScottLocale.ScottLocalesOfAlgebraicDcpos pt fe 𝓤
+
+\end{code}
+
+contains a proof that the Scott locale of any algebraic dcpo is a spectral
+locale.
+
+In this module, we extend this proof by showing that the Scott locale is
+spectral.
+
+\section{Preliminaries}
+
+The following function expresses a list being contained in a given subset.
+
+\begin{code}
+
+_⊆⊆_ : {𝓤 𝓥 : Universe} {X : 𝓤  ̇} → List X → 𝓟 {𝓥} {𝓤} X → 𝓤 ⊔ 𝓥  ̇
+_⊆⊆_ {_} {_} {X} xs U = (x : X) → member x xs → x ∈ U
+
+\end{code}
+
+We define the following predicate that expresses what it means for two elements
+of a DCPO `𝓓` to be “bounded above”.
+
+\begin{code}
+
+bounded-above : (𝓓 : DCPO {𝓤 ⁺} {𝓤}) → ⟨ 𝓓 ⟩∙ → ⟨ 𝓓 ⟩∙ → Ω (𝓤 ⁺)
+bounded-above 𝓓 x y = ∥ upper-bound (binary-family 𝓤 x y) ∥Ω
+ where
+  open Joins (λ a b → a ⊑⟨ 𝓓 ⟩ₚ b)
+
+infix 30 bounded-above
+
+syntax bounded-above 𝓓 x y = x ↑[ 𝓓 ] y
+
+\end{code}
+
+For the proof of spectrality, we will also need the following decidability
+assumption.
+
+\begin{code}
+
+decidability-condition : (𝓓 : DCPO {𝓤 ⁺} {𝓤}) → 𝓤 ⁺  ̇
+decidability-condition 𝓓 = (c d : ⟨ 𝓓 ⟩∙) →
+                            is-compact 𝓓 c → is-compact 𝓓 d →
+                             is-decidable (bounded-above 𝓓 c d holds)
+
+\end{code}
+
+\section{The proof}
+
+As mentioned previously, we assume a couple of things.
+
+  1. The dcpo `𝓓` in consideration is large and locally small.
+  2. It is pointed.
+  3. It has a specified small compact basis.
+  4. It satisfies the aforementioned decidability condition.
+  5. It is bounded complete (which means it is a Scott domain when combined
+     with the algebraicity condition).
+
+\begin{code}
+
+open DefinitionOfBoundedCompleteness
+
+module SpectralScottLocaleConstruction
+        (𝓓    : DCPO {𝓤 ⁺} {𝓤})
+        (hl   : has-least (underlying-order 𝓓))
+        (hscb : has-specified-small-compact-basis 𝓓)
+        (dc   : decidability-condition 𝓓)
+        (bc   : bounded-complete 𝓓 holds)
+        (pe   : propext 𝓤) where
+
+ open ScottLocaleConstruction 𝓓 hscb pe
+
+\end{code}
+
+We denote by `Σ𝓓` the large and locally small Scott locale of the dcpo `𝓓`.
+
+\begin{code}
+
+ open import Locales.ScottLocale.Properties pt fe 𝓤
+ open ScottLocaleProperties 𝓓 hl hscb pe
+
+ Σ[𝓓] : Locale (𝓤 ⁺) 𝓤 𝓤
+ Σ[𝓓] = Σ⦅𝓓⦆
+
+\end{code}
+
+We denote by `(B, β)` the algebraic basis of the pointed dcpo 𝓓 in consideration.
+
+\begin{code}
+
+ B : 𝓤  ̇
+ B = index-of-compact-basis 𝓓 hscb
+
+ β : B → ⟨ 𝓓 ⟩∙
+ β i = family-of-compact-elements 𝓓 hscb i
+
+ open structurally-algebraic
+
+ scb : is-small-compact-basis 𝓓 (family-of-compact-elements 𝓓 hscb)
+ scb = small-compact-basis 𝓓 hscb
+
+ open is-small-compact-basis scb
+
+ ϟ : (b : B) → is-compact 𝓓 (β b)
+ ϟ = basis-is-compact
+
+\end{code}
+
+We define some nice notation for the prop-valued equality of the dcpo `𝓓`.
+
+\begin{code}
+
+ _＝ₚ_ : ⟨ 𝓓 ⟩∙ → ⟨ 𝓓 ⟩∙ → Ω (𝓤 ⁺)
```

Comment:
Maybe this should go into the `DomainTheory` development?

Reply:
That might be a good idea.

Reply:
I realized this function is actually not used anywhere, but I still added it to the `DomainTheory` development.